### PR TITLE
3.4 - Remove risky tests

### PR DESCRIPTION
**Pull request by antograssiot**
PHPUnit 6 is a bit stricter against risky tests definition.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -28,9 +28,9 @@ init:
 install:
   - cd c:\
   - appveyor DownloadFile http://windows.php.net/downloads/releases/archives/php-5.6.0-nts-Win32-VC11-x86.zip -FileName php.zip
-  - 7z x php.zip -oc:\php
+  - 7z x php.zip -oc:\php > nul
   - appveyor DownloadFile https://dl.dropboxusercontent.com/s/euip490d9183jkr/SQLSRV32.cab -FileName sqlsrv.cab
-  - 7z x sqlsrv.cab -oc:\php\ext php*_56_nts.dll
+  - 7z x sqlsrv.cab -oc:\php\ext php*_56_nts.dll > nul
```
Comment (markstory) [inline appveyor.yml]: Is `nul` correct? Looks like it is missing an `l` to my windows noob brain.

File: tests/TestCase/ORM/TableTest.php [hunk 2]
```diff
@@ -4237,21 +4252,25 @@ public function testBelongsToManyFluentInterface()
...
+                ->setFinder('list')
+                ->setProperty('authors')
+                ->setSource($authors)
+                ->setStrategy('select')
+                ->setSaveStrategy('append')
+                ->setThrough('author_articles')
+                ->setJoinType('inner');
+        } catch (\BadMethodCallException $e) {
+            $this->fail('Method chaining should be ok');
+        }
```
Comment (markstory) [inline tests/TestCase/ORM/TableTest.php]: Wouldn't letting the exception bubble up also work?

File: appveyor.yml (hunk 1 above)
Comment (antograssiot) [inline appveyor.yml]: Yes `nul` is correct. And `$null` in Powershell  form what I have seen. At least here it works and hide the output of the unzipping operation

File: tests/TestCase/ORM/TableTest.php (hunk 2 above)
Comment (antograssiot) [inline tests/TestCase/ORM/TableTest.php]: It would work also but an error would be reporting instead of a falling test so I thought  it was  cleaner

**Comment by antograssiot** [comment]
I will investigate why the travis build are failing in the next days

File: tests/TestCase/ORM/TableTest.php (hunk 2 above)
Comment (markstory) [inline tests/TestCase/ORM/TableTest.php]: Either way works for me. Both error and fail are not passing which is what matters most in my mind.

File: appveyor.yml (hunk 1 above)
Comment (markstory) [inline appveyor.yml]: Neat. TIL.

**Comment by markstory** [comment]
@antograssiot Tests didn't fail. Code coverage seg faulted.

**Pull request merged**